processor/redactionprocessor/processor.go

```diff
@@ -79,6 +97,37 @@ func (s *redaction) processResourceSpan(ctx context.Context, rs ptrace.ResourceS
 	}
 }
 
+// processResourceLog processes the log resource and all of its logs and then returns the last
+// view metric context. The context can be used for tests
+func (s *redaction) processResourceLog(ctx context.Context, rl plog.ResourceLogs) {
+	rsAttrs := rl.Resource().Attributes()
+
+	s.processAttrs(ctx, rsAttrs)
+
+	for j := 0; j < rl.ScopeLogs().Len(); j++ {
+		ils := rl.ScopeLogs().At(j)
+		for k := 0; k < rl.ScopeLogs().Len(); k++ {
+			log := ils.LogRecords().At(k)
+			s.processAttrs(ctx, log.Attributes())
+		}
+	}
+}
+
+func (s *redaction) processResourceMetric(ctx context.Context, rm pmetric.ResourceMetrics) {
+	rsAttrs := rm.Resource().Attributes()
+
+	s.processAttrs(ctx, rsAttrs)
+
+	for j := 0; j < rm.ScopeMetrics().Len(); j++ {
+		ils := rm.ScopeMetrics().At(j)
+		s.processAttrs(ctx, ils.Scope().Attributes())
+		for k := 0; k < ils.Metrics().Len(); k++ {
+			metric := ils.Metrics().At(k)
+			s.processAttrs(ctx, metric.Metadata())
```

Comment:
there is no `Attributes` property within the metrics or its data points, but `Metadata` seems to have a similar meaning, so passing on those to the `processAttrs` function

Reply:
See https://github.com/open-telemetry/opentelemetry-proto/blob/e6923b74ccb43c884308b589cc00f972c12b4055/opentelemetry/proto/metrics/v1/metrics.proto#L212-L213, these are not the attributes you are looking for.

You can take inspiration from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/eba20f5bedcd1ceeb43348d2c8e7ca524c0ecc77/processor/attributesprocessor/attributes_metric.go#L64 on how to do this

Reply:
thanks for the link - I will adapt this accordingly then

Reply:
@mx-psi this should be fixed now